packages/node_modules/node-red/red.js

```diff
@@ -130,6 +155,33 @@ try {
     process.exit();
 }
 
+if (parsedArgs.define) {
+    var defs = parsedArgs.define;
+    defs.forEach(function (def) {
+        try {
+            var match = /^(([^=]+)=(.+)|@(.*))$/.exec(def);
+            if (match) {
+                if (!match[4]) {
+                    var val = JSON.parse(match[3]);
+                    setProperty(settings, match[2], val);
```

Comment:
I suggest you use `RED.util.setObjectProperty` rather than add the new `setProperty` function. It does exactly the same work.

```suggestion
                    RED.util.setObjectProperty(settings, match[2], val, true);
```